crates/sui-core/src/authority/authority_store.rs

```diff
@@ -542,6 +543,62 @@ impl AuthorityStore {
         Ok(result)
     }
 
+    pub fn have_received_object_at_version(
+        &self,
+        object_id: &ObjectID,
+        version: VersionNumber,
+        epoch_id: EpochId,
+    ) -> Result<bool, SuiError> {
+        let object_key = (
+            epoch_id,
+            ObjectKey(*object_id, version),
+            MarkerKind::Received,
+        );
+        Ok(self
+            .perpetual_tables
+            .object_per_epoch_marker_table
+            .get(&object_key)?
+            .is_some())
+    }
+
+    pub fn check_receiving_objects(
+        &self,
+        receiving_objects: &[ObjectRef],
+        input_objects_len: usize,
+        protocol_config: &ProtocolConfig,
+        epoch_id: EpochId,
```

Comment:
consider passing in EpochStore here?

Reply:
also i think it may be better to move this code to transaction_input_checker, but its up to you

Reply:
Happy to pass the EpochStore here instead. I'm in favor of keeping this code here however, since the other checks that this is related/similar to (e.g., `check_input_objects` below this) also reside here, so seems a bit better to keep them close to each other.